streaming/base/partition/orig.py

```diff
@@ -126,10 +143,23 @@ def get_partitions_orig(num_samples: int,
     #
     # ids: (physical nodes, samples per rank, ranks per node).
     overflow = ids.shape[1] % ranks_per_node
+    more_samples_than_ranks = ids.shape[1] > ranks_per_node
     if overflow:
         underflow = ranks_per_node - overflow
-        last = ids[:, -ranks_per_node - underflow + 1:-ranks_per_node + 1]
+        if more_samples_than_ranks:
+            last = ids[:, -ranks_per_node - underflow + 1:-ranks_per_node + 1]
+        else:
+            # There are less samples than ranks. Usually, we pad by trying to ensure that the same
+            # samples don't get repeated over and over, but with in this case, we are forced to.
+            warnings.warn(f'Attempting to partition {ids.shape[1]} samples over {ranks_per_node} \
```

Comment:
Same comment here. Any recommendations we can provide?

Reply:
In this case, there's not much to do except check if your dataset does have the number of samples you expected. added that in.